### ISSUE TITLE
Change to logging released in v2.4.141 fails to set level so DEBUG messages appear

### ISSUE THREAD

**Issue by chrisinmtown** [MISP/PyMISP#731]
After an upgrade to v2.4.141 we noticed our log files are a LOT busier, PyMISP is filling them with DEBUG messages like this:
```20210414-05:00:01 DEBUG GET - https://misp/servers/getPyMISPVersion.json
20210414-05:00:01 DEBUG {}
20210414-05:00:01 DEBUG {'User-Agent': 'PyMISP 2.4.140 - Python 3.6', 'Accept-Encoding': 'gzip, deflate', 'Accept': 'application/json', 'Connection': 'keep-alive', 'Authorization': 'key', 'content-type': 'application/json'}
20210414-05:00:02 DEBUG {'version': '2.4.140'}
```
After some experiments I found that PyMISP changed in 2.4.141; the debug messages do not appear when using PyMISP v2.4.140.  I found commit  00ba313eae23e5d1b6842191d62dd4b56014e5e7 which removed the call to set the default logger level:
```
-logger.setLevel(logging.WARNING)
```
Please restore this setLevel, or something similar so DEBUG messages are only logged if you pass the debug=true argument to the PyMISP class's init method which does this:
```
        if debug:
            logger.setLevel(logging.DEBUG)
```
Thanks for listening

Until then we're downgrading from 2.4.141 back to 2.4.140.

**Comment by Rafiot** [comment]
So that was the reason for the change: https://github.com/MISP/PyMISP/issues/716

I'll investigate the proper way to deal with that.

**Comment by Rafiot** [comment]
I tried again, and I do not have the debug logging output if I don't pass `debug=True` to PyMISP, so I have a hard time understanding what's going on for you.

Is it possible the code you're running sets the log level to DEBUG? If that's the case, it makes sense.

**Comment by chrisinmtown** [comment]
Yes, some of our code sets log level debug.  Here's sample code that demonstrates the behavior.  Run with pymisp==2.4.140, no debug output from PyMISP. Run with pymisp==2.4.141, *lots* of debug output from PyMISP. So it's a new behavior in PyMISP.
```
#!/usr/bin/env python3
import logging
import urllib3
from pymisp import PyMISP

logging.basicConfig(level=logging.DEBUG)
urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
misp = PyMISP(
    url='https://misp.my-company.com:48752',
    key='mykeygoeshere',
    ssl=False)
print('Fetching event')
event = misp.get_event(event='224113', pythonify=False)
```

**Comment by Rafiot** [comment]
Yes, I agree it changes the behavior, but PyMISP was overwriting the loglevel you were passing, and that was a bug on PyMISP side: a library must respect the loglevel required by the user.

**Comment by chrisinmtown** [comment]
I disagree. In the old code PyMISP set the level on its logger, the one it fetched using name 'pymisp', definitely did not touch any logger instances created by my code.

**Comment by Rafiot** [comment]
I know you didn't change anything, because PyMISP enforced the loglevel to WARNING, reverting your configuration (DEVUG< set by `logging.basicConfig(level=logging.DEBUG)`). The library changing your config was a mistake.

**Comment by chrisinmtown** [comment]
I'm breaking my head trying to understand your assertion @Rafiot that the PyMISP invocation of set-level on the 'pymisp'-named logging instance could change the level on my logging instance.  I don't think that's right.

Neither of us are changing the root logger.  Both of us are working with local instances.  Is this making any sense?

**Comment by Rafiot** [comment]
`logging.basicConfig(level=logging.DEBUG)` means "please switch everything to loglevel debug in that whole script I'm running" 

PyMISP forcing the loglevel to `WARNING` means that the default you're requesting for your whole script is discarded and silently changed. It is incorrect and that was a mistake I made a few years ago misunderstanding the way logging worked in Python.

In you case, if you only want the `WARNING` loglevel, you should do `logging.basicConfig(level=logging.WARNING)`, and not `logging.basicConfig(level=logging.DEBUG)`.

You are correct that the debug switch in PyMISP is pretty much useless and should probably be removed, but again, it was due to my misunderstanding of the way logging was working in Python: there is no reason to use if/else to call or not call a logger, as it is handled natively by the loglevel.

**Comment by chrisinmtown** [comment]
Independent levels on logging instances is the huge strength of logging libraries. After I set the ROOT logger to allow the lowest level used by any desired call, I can set only the logging instance I need to watch so it passes DEBUG messages (e.g., Python requests) while limiting all other logging instances (e.g., Pymisp) to pass only ERROR messages.

Anyhow, that's a long way of saying that I don't agree with your explanation @Rafiot. As I understand https://docs.python.org/3/library/logging.html calling `logging.basicConfig(level=logging.DEBUG)` only sets the level on the root logger to DEBUG, so it will allow messages at DEBUG or higher priorities to pass. It does *not* switch any levels on the non-root loggers out there; I can still limit my own script loggers to levels like INFO, WARNING, etc.

I am baffled by your statement:

> PyMISP forcing the loglevel to WARNING

I don't think it was doing anything of the kind.  I feel Pymisp v2.4.140 was only setting a sensible level on its own logger, the one named 'pymisp'.

I was hoping to convince you that setting the level on the PyMISP library logger back to WARNING by default was the right choice, and makes the "debug" parameter to the constructor sensible again, but it looks like I have failed to do that.

Now with this v2.4.141 change to PyMISP, to keep the v2.4.140 behavior and avoid undesired output we have to add this line of code to all of our PyMISP scripts:
```
logging.getLogger('pymisp').setLevel(logging.WARNING)
```
To be fair, we only need to do this in our scripts that configure the root logger at level DEBUG -- but we do that in basically all of our scripts.
